### PR TITLE
TOAZ-284 Create Batch Pools through WSM

### PR DESCRIPTION
**Pull request by sergiygetlin**
This PR implements new endpoint to create Azure Batch Pool. Pool will be created in the shared batch account which belongs to a landing zone.

### CODE REVIEW

**Review by giventocode — APPROVED** [review]
Good job! One minor naming comment.

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/batchpool/AzureBatchPoolHelper.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.batchpool;
+
+public class AzureBatchPoolHelper {
```
[the remark this answers is not in the document]
Comment (sergiygetlin) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/batchpool/AzureBatchPoolHelper.java:3]: Yes. Not the name looks too generic. Probably will leave it as is for now. Maybe I incorporate some helpers functionality later or I will refactor it later.

[the remark this answers is not in the document]
Comment (sergiygetlin) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/batchpool/AzureBatchPoolHelper.java:3]: I know we need delete functionality for pool as well. Will consider it as a refactoring for this future task.

**Review by rtitle — APPROVED** [review]
I didn't test it myself but looks good to me overall. Nice work

**Pull request merged**